flang/lib/Lower/DirectivesCommon.h

```diff
@@ -128,6 +128,7 @@ static void processOmpAtomicTODO(mlir::Type elementType,
                              Fortran::parser::OmpAtomicClauseList>()) {
     // Based on assertion for supported element types in OMPIRBuilder.cpp
     // createAtomicRead
+    // TODO: Enumerate remaining unsupported types for atomics
     mlir::Type unwrappedEleTy = fir::unwrapRefType(elementType);
     bool supportedAtomicType = fir::isa_trivial(unwrappedEleTy);
```

Comment:
Should we let this in? Or we are in a position to remove this? I didn't remove right now; thought to ask once.

Reply:
I agree it can be replaced by an assertion

Reply:
I would replace the function with something like `assert(fir::isa_trivial(fir::unwrapRefType(elementType)) && "is supported type for omp atomic")`